rplugin/python3/deoplete/sources/deoplete_jedi/server.py

```diff
@@ -524,12 +524,17 @@ def completions(self, *args):
         restart the server.
         """
         if self._count > self.max_completion_count:
-            self._count = 0
             self.restart()
 
         self._count += 1
-        stream_write(self._server.stdin, args)
-        return stream_read(self._server.stdout)
+        try:
+            stream_write(self._server.stdin, args)
+            return stream_read(self._server.stdout)
+        except StreamError as exc:
+            log.error('Caught %s during handling completions(%s), '
+                      ' restarting server', exc, args)
```

Comment:
Do you recommend a way to test issues in this context?
Just throw an error manually?